File: RUCropController/Views/RUCropScrollView.m
//
//  RUScrollView.m
//  RUCropController
//
//  Created by Roman Ustiantcev on 06/10/2017.
//  Copyright © 2017 Roman Ustiantcev. All rights reserved.
//

#import "RUCropScrollView.h"

@implementation RUCropScrollView

- (void)touchesBegan:(NSSet *)touches withEvent:(UIEvent *)event
{
    if (self.touchesBegan)
        self.touchesBegan();
    
    [super touchesBegan:touches withEvent:event];
}

- (void)touchesEnded:(NSSet *)touches withEvent:(UIEvent *)event
{
    if (self.touchesEnded)
        self.touchesEnded();
    
    [super touchesEnded:touches withEvent:event];
}

- (void)touchesCancelled:(NSSet *)touches withEvent:(UIEvent *)event
{
    if (self.touchesCancelled)
        self.touchesCancelled();
    
    [super touchesCancelled:touches withEvent:event];
}


@end

File: RUCropController/Views/RUCropScrollView.swift
//
//  RUCropScrollView.swift
//  RUCropController
//
//  Created by Roman Ustiantcev on 09/10/2017.
//  Copyright © 2017 Roman Ustiantcev. All rights reserved.
//

import UIKit

@objc class RUCropScrollView: UIScrollView {
    
    @objc var touchesBegan: (() -> Void)?
    @objc var touchesCancelled: (() -> Void)?
    @objc var touchesEnded: (() -> Void)?
    
    @objc override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
        if (self.touchesBegan != nil) {
            self.touchesBegan!()
        }
        super.touchesBegan(touches, with: event)
    }
    
    @objc override func touchesEnded(_ touches: Set<UITouch>, with event: UIEvent?) {
        if (self.touchesEnded != nil) {
            self.touchesEnded!()
        }
        super.touchesEnded(touches, with: event)
    }
    
    @objc override func touchesCancelled(_ touches: Set<UITouch>, with event: UIEvent?) {
        if (self.touchesCancelled != nil) {
            self.touchesCancelled!()
        }
        super.touchesCancelled(touches, with: event)
    }
    
}
